src/xnvme_disk_io.cpp

```diff
@@ -120,15 +120,12 @@ namespace {
 			iovec_t buf = {bufz, r.length};
 
 			// TODO: this function isn't being free()d anywhere right now
-			std::function<void()> *whandler = new std::function<void()>([=, handler = std::move(handler)]() {
+			auto whandler = [=, handler = std::move(handler)](storage_error error) mutable {
```

Comment:
this `=` capture makes me a bit nervous. Does it capture a lot? If you were to explicitly list everything it captures, would that be an unreasonably long list? it seems like it captures `this`

Reply:
Nice catch -- it captures `this`, `bufz, and `start_time`, so I agree it's better to write it out completely.

The capture of `this` gives access to `m_ios` and `m_stats_counters`, and the `libtorrent::buffer_allocator_interface` implemented by `xnvme_disk_io`. 

> it seems like it captures this

I can see it causing trouble if the lifetime of `xnvme_disk_io` is not guaranteed to be longer than the invocation of the callback being declared. Is this (or something else) going to be a potential problem?

Reply:
> I can see it causing trouble if the lifetime of xnvme_disk_io is not guaranteed to be longer than the invocation of the callback being declared. Is this (or something else) going to be a potential problem?

No, I think that's fine

Reply:
Cool!